### PR TITLE
mgr/cephadm: do not accept filestore osds in osdspecs

### PR DESCRIPTION
**Pull request by jschmid1**
Signed-off-by: Joshua Schmid <jschmid@suse.de>

Fixes: https://tracker.ceph.com/issues/45980

## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by jschmid1** [comment]
jenkins retest this please

**Comment by jschmid1** [comment]
jenkins test make check

File: src/python-common/ceph/deployment/drive_group.py [hunk 1]
```diff
@@ -288,8 +288,9 @@ def validate(self):
             if s.all:
                 raise DriveGroupValidationError("`all` is only allowed for data_devices")
 
-        if self.objectstore not in ('filestore', 'bluestore'):
-            raise DriveGroupValidationError("objectstore not in ('filestore', 'bluestore')")
+        if self.objectstore not in ('bluestore'):
+            raise DriveGroupValidationError(f"{self.objectstore} is not supported. Must be "
+                                            f"one of ('bluestore')")
```
Comment (varshar16) [inline src/python-common/ceph/deployment/drive_group.py:293]: nit:
```suggestion
        if self.objectstore != 'bluestore':
            raise DriveGroupValidationError(f"{self.objectstore} is not supported. Only "
                                            " 'BlueStore' is supported.")
```

**Comment by varshar16** [comment]
Also your commit is missing tracker ticket.

**Comment by sebastian-philipp** [comment]
this not only removed filestore form cephadm, but also from all other orchestrators. I'm ok with it. @BlaineEXE + @travisn ?

**Comment by travisn** [comment]
> this not only removed filestore form cephadm, but also from all other orchestrators. I'm ok with it. @BlaineEXE + @travisn ?

Sounds good, rook already removed support for filestore in v1.3

**Comment by sebastian-philipp** [comment]
* https://pulpito.ceph.com/swagner-2020-07-14_12:03:52-rados:cephadm-wip-swagner-testing-2020-07-14-1125-distro-basic-smithi/
* https://pulpito.ceph.com/swagner-2020-07-14_13:50:58-rados:cephadm-wip-swagner-testing-2020-07-14-1125-distro-basic-smithi/

https://tracker.ceph.com/issues/46534

**Pull request merged**